### PR TITLE
Add behaviors

### PR DESCRIPTION
**Pull request by flands**
- Behaviors
- Configuration
- Semantic Conventions

Behavior is new to cover specifics of some subset. Its goal is to drive consistency across GDI repositories.

### CODE REVIEW

File: specification/behaviors.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+MUST be able to associate the call stack to the span. This association SHOULD
+happen as close to the sampling point as feasible, but MAY occur later in a
+processing pipeline.
+
+After the association has been made, the `TraceId` and `SpanId` fields of the
+<<<<<<< HEAD
+[LogRecord](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/logs/v1/logs.proto#L113)
+=======
+LogRecord](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/logs/v1/logs.proto#L113)
+>>>>>>> a62423a (Add behaviors)
```
Comment (iNikem) [inline specification/behaviors.md:46]: Surely you want to resolve this :)

Comment (pellared) [reply specification/behaviors.md]: Addressed https://github.com/signalfx/gdi-specification/pull/157/commits/85f037add667eb29ebaf155ceecfb608893163a1 😉

File: specification/behaviors.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
+# Behaviors
```
Comment (pellared) [inline specification/behaviors.md:1]: 1. Can you add some overall description e.g. like

```
This document describes the specific behaviors for Splunk GDI Components.
```

2. What are other examples of behaviors? RUM?

3. Could we add a reference to this document in the main `README.md`?

File: specification/behaviors.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+## Profiling
+
+**Status**: [Experimental](../README.md#versioning-and-status-of-the-specification)
+
+This document describes the specific behavior and conventions for Splunk
```
Comment (pellared) [inline specification/behaviors.md:7]:
```suggestion
This section describes the behavior for Splunk
```

File: specification/behaviors.md [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+### Call Stack Span Association
+
+When a call stack is sampled during the execution of a span scope, the profiler
+MUST be able to associate the call stack to the span. This association SHOULD
+happen as close to the sampling point as feasible, but MAY occur later in a
+processing pipeline.
+
+After the association has been made, the `TraceId` and `SpanId` fields of the
+[LogRecord message](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/logs/v1/logs.proto)
+MUST be populated (see [below](#logrecord-fields)).
```
Comment (pellared) [inline specification/behaviors.md:43]: this hyperlink looks wrong

I think it should be 

```suggestion
MUST be populated (see [here](configuration.md#logrecord-fields)).
```

File: specification/semantic_conventions.md [hunk 5]
```diff
@@ -82,3 +82,99 @@ Web RUM library MUST add the following attributes to all spans:
 | ----                     | ----   | -----------                                                                                                                    |
 | `location.href`          | string | Value of [`location.href`](https://developer.mozilla.org/en-US/docs/Web/API/Location/href) at the moment of creating the span. |
 | `splunk.script_instance` | string | A 64bit identifier, unique to every instance of the `SplunkRum` library.                                                       |
+
+## Profiling ResourceLogs
```
Comment (pellared) [inline specification/semantic_conventions.md:86]: nit

```suggestion
## Profiling `ResourceLogs` Message
```

File: specification/semantic_conventions.md [hunk 6]
```diff
@@ -82,3 +82,99 @@ Web RUM library MUST add the following attributes to all spans:
...
+
+**Status**: [Experimental](../README.md#versioning-and-status-of-the-specification)
+
+`ResourceLogs` is the [upstream protobuf data
+type](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/logs/v1/logs.proto#L47).
+It MUST be populated with the attributes from the OpenTelemetry resource.
+
+### InstrumentationLibraryLogs
```
Comment (pellared) [inline specification/semantic_conventions.md:94]: nit

```suggestion
### `InstrumentationLibraryLogs` Message
```

File: specification/semantic_conventions.md [hunk 7]
```diff
@@ -82,3 +82,99 @@ Web RUM library MUST add the following attributes to all spans:
...
+
+Each `ResourceLogs` instance has an instance of `InstrumenationLibraryLogs`.
+For each `InstrumentationLibraryLogs` instance:
+
+- `name` - MUST be set to `otel.profiling`
+- `version` - MUST be set to `0.1.0`
+
+### LogRecord Common Attributes
```
Comment (pellared) [inline specification/semantic_conventions.md:102]: nit

```suggestion
### `LogRecord` Messsage Common Attributes
```

File: specification/semantic_conventions.md [hunk 8]
```diff
@@ -82,3 +82,99 @@ Web RUM library MUST add the following attributes to all spans:
...
+
+Inside each `InstrumentationLibraryLogs` instance is a list of `LogRecord`
+instances. For each `LogRecord` instance:
+
+- `com.splunk.sourcetype` MUST be set to the value `otel.profiling`
+- `source.event.period` MUST contain the sampling period for call stack sampler
+- `source.event.name` OPTIONALLY can contain the name of the event that triggered the sampling
+
+### LogRecord Fields
```
Comment (pellared) [inline specification/semantic_conventions.md:111]: nit

```suggestion
### `LogRecord` Message Fields
```

**Review by pellared — APPROVED** [review]

File: specification/behaviors.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+### Call Stack Span Association
+
+When a call stack is sampled during the execution of a span scope, the profiler
+MUST be able to associate the call stack to the span. This association SHOULD
+happen as close to the sampling point as feasible, but MAY occur later in a
+processing pipeline.
+
+After the association has been made, the `TraceId` and `SpanId` fields of the
+[LogRecord message](https://github.com/open-telemetry/opentelemetry-proto/blob/main/opentelemetry/proto/logs/v1/logs.proto)
+MUST be populated (see [here](configuration.md#logrecord-fields)).
```
Comment (pellared) [inline specification/behaviors.md:45]: My bad

```suggestion
MUST be populated (see [here](semantic_conventions.md#logrecord-fields)).
```

**Review by kubawach — APPROVED** [review]

**Pull request merged**